agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/ConnectionString.java

```diff
@@ -39,9 +39,15 @@
 
   private ConnectionString() {}
 
-  public static void parseInto(String connectionString, TelemetryClient targetConfig)
-      throws InvalidConnectionStringException {
-    mapToConnectionConfiguration(getKeyValuePairs(connectionString), targetConfig);
+  public static void parseInto(String connectionString, TelemetryClient telemetryClient)
+      throws InvalidConnectionStringException, MalformedURLException {
+    if (Strings.isNullOrEmpty(connectionString)) {
+      telemetryClient.setInstrumentationKey(null);
+      // reset endpoint to default for ingestion, live metric and profiler
+      telemetryClient.getEndpointProvider().resetEndpointUrls();
```

Comment:
we should do this also in `mapToConnectionConfiguration` below before starting to overwrite them, since we only conditionally overwrite them below

Reply:
in maptoconnectionConfiguration, we should reset all? the above doesn't impact statsbeat and profiler.